### PR TITLE
Bug 1811976 - Hide Sync FAB in TabsTray when user not signed in

### PR DESCRIPTION
**Pull request by gabrielluong**
This carries over https://github.com/mozilla-mobile/firefox-android/pull/3732 with a fix to UI test.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1811976

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/TabDrawerRobot.kt [hunk 1]
```diff
@@ -297,12 +296,6 @@ class TabDrawerRobot {
             itemContainingText(getStringResource(R.string.synced_tabs_sign_in_message)),
             itemContainingText(getStringResource(R.string.sync_sign_in)),
         )
-        assertItemWithResIdAndDescriptionExists(
-            itemWithResIdAndDescription(
-                "$packageName:id/new_tab_button",
-                getStringResource(R.string.resync_button_content_description),
-            ),
-        )
     }
```
Comment (Amejia481) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/TabDrawerRobot.kt:299]: Just want to confirm, do we were expecting that https://bugzilla.mozilla.org/show_bug.cgi?id=1811976 was happening?

**Review by Amejia481 — APPROVED** [review]
In general this looks good, just one curiosity, added a comment to clarify.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/TabDrawerRobot.kt (hunk 1 above)
Comment (gabrielluong) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/TabDrawerRobot.kt:299]: This UI test assertion was confirming that the bug was happening. We don't want the "Sync" button to be displayed when the user is not signed into FxA

Comment (Amejia481) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/TabDrawerRobot.kt:299]: Thanks for validating :)

**Pull request merged**